lib/EioDriver.php

```diff
@@ -356,11 +356,39 @@ private function onUnlink($data, $result, $req) {
     /**
      * {@inheritdoc}
      */
-    public function mkdir($path, $mode = 0644) {
+    public function mkdir($path, $mode = 0644, $recursive) {
         \call_user_func($this->incrementor, 1);
         $promisor = new Deferred;
         $priority = \EIO_PRI_DEFAULT;
-        \eio_mkdir($path, $mode, $priority, [$this, "onGenericResult"], $promisor);
+
+        if ($recursive) {
+            $arrayPath = array_filter(explode(DIRECTORY_SEPARATOR, $path));
```

Comment:
I guess we should first replace all `/` with `DIRECTORY_SEPARATOR`, otherwise `/` won't work on Windows.